sdk/communication/communication-network-traversal/test/public/utils/recordedClient.ts

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Context } from "mocha";
+import * as dotenv from "dotenv";
+
+import {
+  env,
+  Recorder,
+  record,
+  RecorderEnvironmentSetup,
+  isPlaybackMode
+} from "@azure/test-utils-recorder";
+import {
+  DefaultHttpClient,
+  HttpClient,
+  HttpOperationResponse,
+  isNode,
+  TokenCredential,
+  WebResourceLike
+} from "@azure/core-http";
+import { CommunicationRelayClient, CommunicationRelayClientOptions } from "../../../src";
+import { ClientSecretCredential, DefaultAzureCredential } from "@azure/identity";
+import { parseConnectionString } from "@azure/communication-common";
+
+if (isNode) {
+  dotenv.config();
+}
+
+export interface RecordedClient<T> {
+  client: T;
+  recorder: Recorder;
+}
+
+const replaceableVariables: { [k: string]: string } = {
+  COMMUNICATION_CONNECTION_STRING: "endpoint=https://endpoint/;accesskey=banana",
+  INCLUDE_PHONENUMBER_LIVE_TESTS: "false",
+  COMMUNICATION_ENDPOINT: "https://endpoint/",
+  AZURE_CLIENT_ID: "SomeClientId",
+  AZURE_CLIENT_SECRET: "SomeClientSecret",
+  AZURE_TENANT_ID: "SomeTenantId"
+};
+
+export const environmentSetup: RecorderEnvironmentSetup = {
+  replaceableVariables,
+  customizationsOnRecordings: [
+    (recording: string): string =>
+      recording.replace(/"token"\s?:\s?"[^"]*"/g, `"token":"sanitized"`),
+    (recording: string): string =>
+      recording.replace(/"access_token"\s?:\s?"[^"]*"/g, `"access_token":"sanitized"`),
+    (recording: string): string => recording.replace(/(https:\/\/)([^/',]*)/, "$1endpoint"),
+    (recording: string): string => recording.replace(/"id"\s?:\s?"[^"]*"/g, `"id":"sanitized"`),
+    (recording: string): string => {
+      return recording.replace(
+        /(https:\/\/[^/',]*\/identities\/)[^/',]*(\/token)/,
```

Comment:
you'll need to tailor these for sanitizing the requests you want as these look to be the ones for identity